docs/resources/ui-building-blocks/pages/page-elements.md

```diff
@@ -0,0 +1,414 @@
+| **Selected Icon Color**   | Color Wheel | Specifies the color of the icons when they are selected.                                |
+| **Unselected Icon Color** |  Color Wheel | Specifies the color of the icons when they are not selected.                            |
+
+
+#### Google Nav Bar
+
+This modern Google-style NavBar features a subtle animation that reveals the item label (page name)
+but only displays the label for the selected item.
+
+![nav-bar-google.png](..%2Fimgs%2Fnav-bar-google.png)
+
+**Styling Properties**
+
+- **Nav Bar Color:** Sets the background color of the NavBar.
+- **Selected Icon & Text Color:** Changes the color of the icon and text when an item is selected.
+- **Unselected Icon & Text Color:** Sets the color for icons and text when an item is not selected.
+- **Selected Background Color**: Alters the background color of the selected item.
+- **Show Unselected Border**: Toggles the visibility of a border around unselected items
+- **Border Width:** Specifies the width of the border around the NavBar item buttons.
+- **Border Radius:** Determines the corner roundness of the NavBar item buttons.
+- **Border Color:** Alters the color of the borders around NavBar item buttons.
+- **Nav Button Padding:** Adjusts the padding inside each nav button.
+- **Nav Button Margin:** Controls the margin around each nav button.
+- **Nav Button Alignment:** Allows customization of how nav buttons align within the NavBar. Options
+  like
+  center, space-between, etc., are given.
+- **Gap Between Icon and Text:** Specifies the spacing between the icon and text within nav buttons.
+- **Animation Duration (ms):** Defines how long animations take when switching between items.
+- **Haptic Feedback:** A toggle that enables or disables haptic feedback when interacting with
+  NavBar items, enhancing the tactile experience.
+
+#### Floating Nav Bar
+
+This NavBar style appears as a floating element above the pages and shows labels for all items
+present in the NavBar.
+
+![nav-bar-floating.png](..%2Fimgs%2Fnav-bar-floating.png)
+
+**Styling Properties**
+
+- **Nav Bar Color:** Sets the background color of the NavBar.
+- **Selected Icon & Text Color:** Specifies the color of the icon and text when an item is selected.
+- **Unselected Icon & Text Color:** Defines the color for the icons and text when they are not
+  selected.
+- **Selected Background Color:** Alters the background color of the selected item.
+- **Width:** Controls the width of the NavBar.
+- **Border Radius:** Determines the roundness of the NavBar's corners.
+- **Elevation:** Adjusts the shadow or elevation effect beneath the NavBar, which helps give
+  the NavBar a floating appearance above other content.
+- **Button Border Radius:** Specifies the radius for the borders of each button within the NavBar.
+- **Nav Button Margin:** Sets the margin around each nav button
+- **Nav Button Padding:** Controls the padding inside each nav button.
```

Comment:
Maybe we could add [this](https://youtu.be/Qhe8X5ykK54) youtube video after this?

Reply:
Great catch! Added